### PR TITLE
[1/X] DXCDT-441: Reintroduce support for aws client addon

### PR DESCRIPTION
**Pull request by sergiught**
### 🔧 Changes

Reintroducing support for aws client addon on the v1 branch after correctly typing them to the new go-auth0 types.

### 📝 Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: internal/auth0/client/resource.go [hunk 1]
```diff
@@ -481,6 +481,42 @@ func NewResource() *schema.Resource {
...
+							Description: "AWS Addon configuration.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"principal": {
+										Description: "AWS principal ARN, e.g. `arn:aws:iam::010616021751:saml-provider/idpname`.",
+										Type:        schema.TypeString,
+										Optional:    true,
+									},
+									"role": {
+										Description: "AWS role ARN, e.g. `arn:aws:iam::010616021751:role/foo`.",
```
Comment (Widcket) [inline internal/auth0/client/resource.go:504]:
```suggestion
										Description: "AWS role ARN, for example `arn:aws:iam::010616021751:role/foo`.",
```

File: internal/auth0/client/resource.go [hunk 2]
```diff
@@ -481,6 +481,42 @@ func NewResource() *schema.Resource {
...
+					Schema: map[string]*schema.Schema{
+						"aws": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							MaxItems:    1,
+							Description: "AWS Addon configuration.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"principal": {
+										Description: "AWS principal ARN, e.g. `arn:aws:iam::010616021751:saml-provider/idpname`.",
```
Comment (Widcket) [inline internal/auth0/client/resource.go:499]:
```suggestion
										Description: "AWS principal ARN, for example `arn:aws:iam::010616021751:saml-provider/idpname`.",
```

Comment (Widcket) [reply internal/auth0/client/resource.go:499]: The docs team's guidelines instruct not to use latin abbreviations.

File: internal/auth0/client/expand.go [hunk 3]
```diff
@@ -236,6 +236,44 @@ func expandClientMetadata(d *schema.ResourceData) *map[string]interface{} {
...
+	return &addons
+}
+
+func expandClientAddonAWS(awsCfg cty.Value) *management.AWSClientAddon {
+	var awsAddon management.AWSClientAddon
+
+	awsCfg.ForEachElement(func(_ cty.Value, awsCfg cty.Value) (stop bool) {
+		awsAddon.Principal = value.String(awsCfg.GetAttr("principal"))
+		awsAddon.Role = value.String(awsCfg.GetAttr("role"))
+		awsAddon.LifetimeInSeconds = value.Int(awsCfg.GetAttr("lifetime_in_seconds"))
```
Comment (Widcket) [inline internal/auth0/client/expand.go:265]: 656I see this was updated in #656 https://github.com/auth0/terraform-provider-auth0/blob/52cdcf0a944dbfd714af2179573305ece61b8961/internal/auth0/client/expand.go#L263

File: internal/auth0/client/resource.go (hunk 2 above)
Comment (sergiught) [reply internal/auth0/client/resource.go:499]: These are exactly the ones taken from https://auth0.com/docs/api/management/v2, however I agree that `for example` reads better.

File: internal/auth0/client/expand.go (hunk 3 above)
Comment (sergiught) [reply internal/auth0/client/expand.go:265]: Yeah, I was playing around with a new style in the second PR to try to circumvent the API behavior when you switch from one addon to another one. Both are equivalent but I think we'll go forward with the ones from the second PR.

**Pull request merged**